backend/app/services/adapters/task_kinds/operations.py

```
@@ -0,0 +1,947 @@
+            and task_crd.metadata.labels.get("autoDeleteExecutor") == "true"
+        ):
+            raise HTTPException(
+                status_code=400,
+                detail="task already clear, please create a new task",
+            )
+
+        # Check expiration
+        expire_hours = settings.APPEND_CHAT_TASK_EXPIRE_HOURS
+        task_type = (
+            task_crd.metadata.labels
+            and task_crd.metadata.labels.get("taskType")
```

Ruff (0.14.13)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)